gladier/state_models.py

```diff
@@ -135,7 +171,12 @@ def next(
             self.next_state = next_state
         return self
 
-    def get_child_states(self) -> t.List[GladierBaseState]:
+    def get_child_states(self) -> t.List[AWSBaseState]:
+        """
+        Get child states.
+
+        TODO: It isn't clear to me why this may be raising attribute errors.
```

Comment:
The attribute `self.next_state` will only exist if `next()` has been called as it is not initialized in the `__init__`. So, if  there's a call to `get_child_states` on an instance for which `next()` has not been called, an `AttributeError` would be raised.

Reply:
I see, so we never assume self.next will ever be defined. Is there a downside to sticking `self.next = None` in a constructor and checking there instead? I think that would read better for me.

Reply:
I feel like there was probably a reason I didn't do this as I don't like this current syntax better either, but I'm not sure what it was now. Pydantic can be somewhat picky about setting attrs on a model class instance, but since we're doing that here anyway, I don't think that is it. If you'd like, I can checkout this branch and give it a try.

Reply:
Fixed in #273